src/Expr.cc

```diff
@@ -1577,26 +1577,41 @@ ValPtr SizeExpr::Fold(Val* v) const
 	return v->SizeVal();
 	}
 
-AddExpr::AddExpr(ExprPtr arg_op1, ExprPtr arg_op2)
-	: BinaryExpr(EXPR_ADD, std::move(arg_op1), std::move(arg_op2))
+// Fill op1 and op2 type tags into bt1 and bt2.
+//
+// If both operands are vectors, use their yield type tag. If
+// either, but not both operands, is a vector, cause an expression
+// error and return false.
+static bool types_from_scalars_or_vectors(Expr* e, TypeTag& bt1, TypeTag& bt2)
```

Comment:
How about prefixing the name of this with `get_`? I don't think it's obvious what it does just from the name as-is.

Reply:
Thanks, I've added the `get_`.